### PR TITLE
Non customer comms

### PR DESCRIPTION
**Pull request by fleetwoods**
Handle TriggeredV3 events (as well as existing TriggeredV2 events) and deal with the new DeliverTo field.

Handle CancellationRequested and CancellationRequestedV2 events.

### CODE REVIEW

File: src/main/scala/com/ovoenergy/orchestration/domain/customer.scala [hunk 1]
```diff
@@ -11,12 +12,22 @@ package object customer {
...
+  case class CustomerProfileResponse(name: CustomerProfileName,
+                                     emailAddress: Option[String],
+                                     phoneNumber: Option[String],
+                                     communicationPreferences: Seq[CommunicationPreference]) {
+    def toCustomerProfile = CustomerProfile(name, ContactProfile(emailAddress, phoneNumber, communicationPreferences))
+  }
+
+  case class ContactProfile(emailAddress: Option[String],
+                            phoneNumber: Option[String],
+                            communicationPreferences: Seq[CommunicationPreference])
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/domain/customer.scala]: Shouldn't `communicationPreferences` be in `CustomerProfile`? Non-customers don't have any comm prefs.

File: src/main/scala/com/ovoenergy/orchestration/domain/customer.scala [hunk 2]
```diff
@@ -11,12 +12,22 @@ package object customer {
...
+  case class CustomerProfile(name: CustomerProfileName, contactProfile: ContactProfile) {
+    def toModel = model.CustomerProfile(
+      firstName = name.firstName,
+      lastName = name.lastName
+    )
+  }
 
-  // DeliverTo is either phone number or email address
-  case class CustomerDeliveryDetails(name: CustomerProfileName, deliverTo: String)
+  case class CustomerProfileResponse(name: CustomerProfileName,
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/domain/customer.scala]: This file seems to contain a mix of classes that are actually domain models for the orchestrator service, and classes that are used to deserialise the JSON response from the profiles service. Could we split these into separate files/packages?

File: src/main/scala/com/ovoenergy/orchestration/kafka/OrchestratedEmailEvent.scala [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
 import scala.concurrent.Future
 
 object OrchestratedEmailEvent {
 
-  def send(sendEvent: OrchestratedEmailV2 => Future[RecordMetadata]) = {
-    (customerDeliveryDetails: CustomerDeliveryDetails, triggered: TriggeredV2, internalMetadata: InternalMetadata) =>
-      val orchestratedEmailEvent = OrchestratedEmailV2(
-        metadata = Metadata.fromSourceMetadata(
+  def send(sendEvent: OrchestratedEmailV3 => Future[RecordMetadata]) = {
+    (customerProfile: Option[CustomerProfile], deliverTo: String, triggered: TriggeredV3,
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/kafka/OrchestratedEmailEvent.scala]: Some of these function signatures are getting a bit out of hand, aren't they? I think orchestrator is ripe for a bit of refactoring, especially as it is one of our most complex services these days. But let's save refactoring for a separate PR.

File: src/main/scala/com/ovoenergy/orchestration/kafka/OrchestratedSMSEvent.scala [hunk 4]
```diff
@@ -1,30 +1,25 @@
...
+    (customerProfile: Option[CustomerProfile], deliverTo: String, triggered: TriggeredV3,
+     internalMetadata: InternalMetadata) =>
+      val orchestratedSMSEvent = OrchestratedSMSV2(
+        metadata = MetadataV2.fromSourceMetadata("orchestration", triggered.metadata),
+        customerProfile = customerProfile,
         templateData = triggered.templateData,
         internalMetadata = internalMetadata,
         expireAt = triggered.expireAt,
-        recipientPhoneNumber = customerProfile.deliverTo
+        recipientPhoneNumber = deliverTo
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/kafka/OrchestratedSMSEvent.scala]: I'd like to see proper domain models for phone numbers and email addresses. Currently they are being passed around as strings. Again, this refactoring could be part of a separate PR.

File: src/main/scala/com/ovoenergy/orchestration/scheduling/package.scala [hunk 5]
```diff
@@ -21,28 +22,82 @@ package object scheduling {
...
+          case Customer(customerId) => Some(customerId)
+          case _                    => None
+        },
+        commName = triggeredV3.metadata.commManifest.name,
         orchestrationExpiry = Instant.now(),
         history = Seq.empty[Change]
       )
     }
+
+    def cancellationRequestedToV2(cancellationRequested: CancellationRequested): CancellationRequestedV2 = {
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/scheduling/package.scala]: `Schedule` seems like a strange place for these conversion functions to live

File: src/main/scala/com/ovoenergy/orchestration/kafka/Producer.scala [hunk 6]
```diff
@@ -14,25 +15,28 @@ import scala.concurrent.Future
...
+      hasCommName: HasCommName[A]): A => Future[RecordMetadata] = {
 
     implicit val scheduler = actorSystem.scheduler
 
     val producer = KafkaProducer(Conf(new StringSerializer, serialiser, hosts))
 
     (event: A) =>
       {
-        logInfo(hasids.traceToken(event), s"Posting event to $topic")
+        logDebug(event, s"Posting event to $topic")
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/orchestration/kafka/Producer.scala]: +1, should this be info rather than debug?

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 228](https://circleci.com/gh/ovotech/comms-orchestration/228) at 2017-05-22 16:12 BST.
